### PR TITLE
Add user_models_generator option to Algo and AggregateAlgo

### LINKED ISSUE #32: Algos: do not load list of models in memory
We could use a generator to load them when needed, it will be needed to avoid running out of memory.

To be updated:
- traintuple.models
- aggregatetuple.models

### PR DESCRIPTION
**Pull request by jmorel**
Fixes #32 

Adds a `user_models_generator` that sets whether to pass models as list or as generators to the train or aggregate method.

### CODE REVIEW

File: tests/test_algo.py [hunk 1]
```diff
@@ -166,3 +167,27 @@ def test_execute_predict(workdir, create_models):
     with open(pred_path, 'r') as f:
         pred = json.load(f)
     assert pred == 'XXX'
+
+
+def test_models_generator(mocker, workdir, create_models):
```
Comment (samlesu) [inline tests/test_algo.py:172]: Could be factorized using two parameters: `use_generator`, `model_argument_type` ?

File: tests/test_algo.py [hunk 2]
```diff
@@ -166,3 +167,27 @@ def test_execute_predict(workdir, create_models):
...
+    _, model_filenames = create_models
+
+    command = ['train']
+    command.extend(model_filenames)
+
+    a1 = DummyAlgo()
+    mocker.patch.object(a1, 'train', autospec=True, return_value={})
```
Comment (samlesu) [inline tests/test_algo.py:179]: 👍  that's a really good way to test the type of the input models.

**Comment by samlesu** [comment]
This should be fully compatible with existing algos and existing backend version.

File: tests/test_algo.py (hunk 1 above)
Comment (jmorel) [reply tests/test_algo.py:194]: I could but I would lose the final `assert models_1 == list(models_2)`

Comment (samlesu) [reply tests/test_algo.py:194]: I think this check is not required, the goal of this test is to check the type of the models argument.

If we really want to do this check we could check directly the value of `models`. It looks `create_models` method returns as first  arg the values inside the models.

Comment (jmorel) [reply tests/test_algo.py]: Ok, I parametrized the test, can you re-review?

**Pull request merged**